sdk/identity/Azure.Identity/src/Credentials/EnvironmentCredentialOptions.cs

```diff
@@ -13,45 +13,45 @@ public class EnvironmentCredentialOptions : TokenCredentialOptions, ISupportsDis
         /// <summary>
         /// The ID of the tenant to which the credential will authenticate by default. This value defaults to the value of the environment variable AZURE_TENANT_ID.
         /// </summary>
-        public string TenantId { get; set; } = EnvironmentVariables.TenantId;
+        internal string TenantId { get; set; } = EnvironmentVariables.TenantId;
```

Comment:
The kind of weird thing with disabling most options which we can set via the environment is that `AdditionallyAllowedTenants` is also setable through the environment `AZURE_ADDITIONALLY_ALLOWED_TENANTS` but is still public, and so is `AuthoritityHost` from the base class `AZURE_AUTHORITY_HOST`. Perhaps, this doesn't matter, but it seems like we're making some odd distinctions.

Reply:
discussed offline and we'll keep it as is.